### DISCUSSION TITLE
Auto-populate user details from SSO provider

### DISCUSSION THREAD

**Discussion by in2uitions** [directus/directus#17321] [Feature Requests]
### Describe the Bug

Implemented SSO like currently documented in directus SSO Facebook example

### To Reproduce

AUTH_FACEBOOK_RESPONSE="tokens,public_profile,jwt,json"
AUTH_FACEBOOK_CLIENT_ID="xxx" 
AUTH_FACEBOOK_CLIENT_SECRET="xxx"
AUTH_FACEBOOK_DRIVER="oauth2"
AUTH_FACEBOOK_SCOPE="email"
AUTH_FACEBOOK_AUTHORIZE_URL="https://www.facebook.com/dialog/oauth"
AUTH_FACEBOOK_ACCESS_URL="https://graph.facebook.com/oauth/access_token"
AUTH_FACEBOOK_PROFILE_URL="https://graph.facebook.com/me?fields=email"
AUTH_FACEBOOK_ICON="facebook" 
AUTH_FACEBOOK_ALLOW_PUBLIC_REGISTRATION="true"
AUTH_FACEBOOK_DEFAULT_ROLE_ID="xxx"

**Comment by rijkvanzanten** [comment]
It's not currently built to do so. There's on standard response format for across every SSO provider, so there's no way to have a single implementation that works for all. We're currently looking to enable hooks at those registration points so you could use Flows or Hooks to fill in the blanks.

**Reply by aidenfoxx** [reply]
There are some fields that are configurable by the user, as per the docs:

```
AUTH_<PROVIDER>_EMAIL_KEY
AUTH_<PROVIDER>_FIRST_NAME_KEY
AUTH_<PROVIDER>_LAST_NAME_KEY
```

They can be used to pick the correct data from the response, but for anything more, the new hook would be required. Reference for the config values: https://docs.directus.io/self-hosted/config-options.html#oauth-2-0

OpenID on the other hand is able to correctly select these values, as they are defined as part of the OpenID spec.

**Reply by in2uitions** [reply]
Thank you 
can you show an example of how to set these attributes from Facebook?
I tried something like this but of course it did not work:
**AUTH_FACEBOOK_LAST_NAME_KEY="https://graph.facebook.com/me?fields=last_name"**
Facebook places these in the public_profile I think but how to access that?
Thank you

**Reply by aidenfoxx** [reply]
Looking at the Facebook API it will probably be:
```
AUTH_<PROVIDER>_FIRST_NAME_KEY="first_name"
AUTH_<PROVIDER>_LAST_NAME_KEY="last_name"
```

https://developers.facebook.com/docs/graph-api/reference/user/#default-public-profile-fields

**Reply by in2uitions** [reply]
not working
Thanks anyway, much appreciated

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
